schema-engine/connectors/sql-schema-connector/src/flavour/postgres/connection.rs

```diff
@@ -5,27 +5,30 @@ use indoc::indoc;
 use psl::PreviewFeature;
 use quaint::{
     connector::{self, tokio_postgres::error::ErrorPosition, PostgresUrl},
-    prelude::{ConnectionInfo, NativeConnectionInfo, Queryable},
+    prelude::{ConnectionInfo, Queryable},
 };
 use schema_connector::{ConnectorError, ConnectorResult, Namespaces};
 use sql_schema_describer::{postgres::PostgresSchemaExt, SqlSchema};
 use user_facing_errors::{schema_engine::ApplyMigrationError, schema_engine::DatabaseSchemaInconsistent};
 
 use crate::sql_renderer::IteratorJoin;
 
+use super::MigratePostgresUrl;
+
 pub(super) struct Connection(connector::PostgreSql);
 
 impl Connection {
     pub(super) async fn new(url: url::Url) -> ConnectorResult<Connection> {
-        let url = PostgresUrl::new(url).map_err(|err| {
-            ConnectorError::user_facing(user_facing_errors::common::InvalidConnectionString {
-                details: err.to_string(),
-            })
-        })?;
+        let url = MigratePostgresUrl::new(url)?;
 
-        let quaint = connector::PostgreSql::new(url.clone())
-            .await
-            .map_err(quaint_err(&url))?;
+        let quaint = match url.0 {
+            PostgresUrl::Native(ref native_url) => connector::PostgreSql::new(native_url.as_ref().clone())
+                .await
+                .map_err(quaint_err(&url))?,
+            PostgresUrl::WebSocket(ref ws_url) => connector::PostgreSql::new_with_websocket(ws_url.clone())
+                .await
+                .map_err(quaint_err(&url))?,
+        };
```

Comment:
Nit: can we factor the duplicated `.await` and `.map_err(quaint_err(&url))?` out of the `match` expression?

E.g.:

```suggestion
        let quaint = match url.0 {
            PostgresUrl::Native(ref native_url) => connector::PostgreSql::new(native_url.as_ref().clone())
            PostgresUrl::WebSocket(ref ws_url) => connector::PostgreSql::new_with_websocket(ws_url.clone())
        }
          .await
          .map_err(quaint_err(&url))?,
```

Reply:
Not the `await` part (different match arms return different futures), but for `map_err` we can